lib/views/staging-view.js

```diff
@@ -391,20 +391,60 @@ export default class StagingView {
         await this.showMergeConflictFileForPath(selectedItem.filePath, {activate: true});
       }
     } else {
-      const pendingItem = this.getPendingFilePatchItem();
-      if (openNew || pendingItem) {
-        const activate = pendingItem === this.props.workspace.getActivePaneItem();
-        await this.showFilePatchItem(selectedItem.filePath, this.selection.getActiveListKey(), {activate});
+      if (openNew) {
+        // User explicitly asked to view diff, such as via click
+        await this.showFilePatchItem(selectedItem.filePath, this.selection.getActiveListKey(), {activate: false});
+      } else {
+        const panesWithStaleItemsToUpdate = this.getPanesWithStalePendingFilePatchItem();
+        if (panesWithStaleItemsToUpdate.length > 0) {
+          // Staging event caused selection to change, so we need to update all old diff views to reflect new selection
+          await Promise.all(panesWithStaleItemsToUpdate.map(async pane => {
+            await this.showFilePatchItem(selectedItem.filePath, this.selection.getActiveListKey(), {
+              activate: false,
+              pane,
+            });
+          }));
+        } else {
+          // Selection was changed via keyboard navigation, update pending item in active pane
+          const activePane = this.props.workspace.getCenter().getActivePane();
+          const activePendingItem = activePane.getPendingItem();
+          const activePaneHasPendingFilePatchItem = activePendingItem && activePendingItem.getRealItem &&
+            activePendingItem.getRealItem() instanceof FilePatchController;
+          if (activePaneHasPendingFilePatchItem) {
+            await this.showFilePatchItem(selectedItem.filePath, this.selection.getActiveListKey(), {
+              activate: false,
+              pane: activePane,
+            });
+          }
+        }
       }
     }
   }
 
-  async showFilePatchItem(filePath, stagingStatus, {activate} = {activate: false}) {
+  getPanesWithStalePendingFilePatchItem() {
+    // "stale" meaning there is no longer a changed file associated with it due to contents being fully staged/unstaged
+    return this.props.workspace.getPanes().filter(pane => {
+      const pendingItem = pane.getPendingItem();
+      if (!pendingItem) { return false; }
+      const isDiffViewItem = pendingItem.getRealItem && pendingItem.getRealItem() instanceof FilePatchController;
+      const isInActiveRepo = pendingItem.getWorkingDirectory() === this.props.workingDirectoryPath;
```

Comment:
add comment `Diff views from repositories other than the active one do not count as stale. Ex: ...`